### PR TITLE
Adds HBBTV extensions to the configuration object

### PR DESCRIPTION
**Pull request by vodlogic**
* Adds subtitlesEnabled property
* Adds audioDescriptionEnabled property
* Adds dtt_network_ids property
* Adds deviceId property
* Adds requestAccessToDistinctiveIdentifier function

### CODE REVIEW

File: types/Configuration.d.ts [hunk 1]
```diff
@@ -86,6 +86,54 @@ declare namespace OIPF {
          */
         pvrEndPadding: number;
 
+        /**
+         * Shall be set to false if subtitles are disabled by the terminal. When set to false, subtitle components that
+         * are selected using a video/broadcast object, A/V control object or HTML5 media element will not be presented.
+         * See also clause 10.2.7 in HBBTV Spec
+         */
+        subtitlesEnabled: boolean;
```
Comment (rathermarv) [inline types/Configuration.d.ts:94]: According the spec these are read-only properties. I can't see a downside off-hand to reflecting that here - if you can't either, could you add that?

**Review by rathermarv — CHANGES_REQUESTED** [review]

File: types/Configuration.d.ts (hunk 1 above)
Comment (vodlogic) [reply types/Configuration.d.ts:94]: Yeah i thought about that but none of the others had it so didnt include. Can add to others aswell?

File: types/Configuration.d.ts [hunk 2]
```diff
@@ -114,5 +162,29 @@ declare namespace OIPF {
...
+         * block while the UI is displayed.
+         * 
+         * If the deviceId property contains a different status code, the terminal shall not display the native UI and
+         * shall immediately call the callback with false as the first argument.
+         * 
+         * If the deviceId property already contains a distinctive identifier, the terminal shall not display the native
+         * UI and shall immediately call the callback with true as the first argument.
+         */
+
+        requestAccessToDistinctiveIdentifier(callback: Function): void;
```
[the remark this answers is not in the document]
Comment (vodlogic) [reply types/Configuration.d.ts:187]: sure

File: types/Configuration.d.ts (hunk 1 above)
Comment (rathermarv) [reply types/Configuration.d.ts:94]: I had a quick look at the OIPF spec, and it _doesn't_ document them as read-only, even though I bet they are.
I also feel a little disquieted about the inconsistency between the two sets and hesitated before leaving the comment.
My feeling was we should just reflect what the docs say. I think we can either make these new props readonly, or not, but leave the others as they are.
Happy with whatever you decide to do with that.

**Review by rathermarv — APPROVED** [review]
Changes made

**Pull request merged**